.eslintrc.js

```diff
@@ -15,6 +15,28 @@ module.exports = {
   extends: ["@elastic/eslint-config-kibana", "plugin:@elastic/eui/recommended"],
   rules: {
     // "@osd/eslint/require-license-header": "off"
+    "import/no-default-export": "off",
+    "@typescript-eslint/naming-convention": [
+      "error",
+      {
+        selector: "default",
+        format: ["camelCase", "UPPER_CASE", "PascalCase", "snake_case"],
+        leadingUnderscore: "allow",
+        trailingUnderscore: "allow",
+      },
+    ],
+    "@osd/eslint/no-restricted-paths": [
+      "error",
+      {
+        basePath: __dirname,
+        zones: [
+          {
+            target: ["(public|server)/**/*"],
+            from: ["../../packages/**/*", "packages/**/*"],
+          },
+        ],
+      },
+    ],
```

Comment:
Not sure about the idea behind this rule, can you add a message that can explain, like here https://github.com/opensearch-project/OpenSearch-Dashboards/blame/791210de027467f95b13f7e0c9582fa149e08eb5/.eslintrc.js#L277

Reply:
@bowenlan-amzn 
The `"@osd/eslint/no-restricted-paths"` rule in the ESLint configuration was added to enforce module import restrictions within the codebase.
`"@typescript-eslint/naming-convention"` rule: it was added to enforce consistent naming conventions across the codebase. This rule specifies that all identifiers (like variables, classes, interfaces, etc.) should follow one of the defined formats: camelCase, UPPER_CASE, PascalCase, or snake_case. It also allows for leading and trailing underscores.

Please let me know if you want me to remove any/both of these rules

Reply:
yes, for `@osd/eslint/no-restricted-paths` part, I suppose it's restricting import from packages in public or server. I am just not sure what's the ideo behind this. @kohinoor98